### PR TITLE
feat: [CDS-40651]: Add checkbox, input for Kustomize Yaml Path

### PR DESCRIPTION
**Pull request by rameshx**
### Summary

Shows a checkbox under Advanced section when `NG_OPTIMIZE_FETCH_FILES_KUSTOMIZE` FF is enabled
Shows an input for entering Kustomize Yaml Path when the checkbox is selected

#### Screenshots

https://user-images.githubusercontent.com/52284933/181499997-fc646a6f-c5f6-4b98-8ec4-d14903ace956.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by rameshx** [comment]
retrigger titlecheck

**Comment by rameshx** [comment]
dockerize

**Comment by rameshx** [comment]
trigger cypress

**Review by YounixIO — DISMISSED** [review]

File: src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/KustomizeWithGIT/__tests__/KustomizeWithGIT.test.tsx [hunk 1]
```diff
@@ -173,4 +175,59 @@ describe('Kustomize with Git/ Github/Gitlab/Bitbucket tests', () => {
...
+        <KustomizeWithGIT {...props} initialValues={initialValues} />
+      </TestWrapper>
+    )
+
+    await act(async () => userEvent.click(getByTestId('advancedTitle-summary')))
+    await act(async () =>
+      userEvent.click(container.querySelector('input[name="optimizedKustomizeManifestCollection"]')!)
+    )
+
+    expect(container.querySelector('input[name="kustomizeYamlFolderPath"]')).toBeInTheDocument()
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/KustomizeWithGIT/__tests__/KustomizeWithGIT.test.tsx:231]: use queryByAttribute

**Review by chetanvekariya-harness — DISMISSED** [review]
Better to address small comments.

File: src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/KustomizeWithGIT/KustomizeWithGIT.tsx [hunk 2]
```diff
@@ -377,34 +402,72 @@ function KustomizeWithGIT({
...
+                            label={getString('pipeline.manifestType.optimizedKustomizeManifestCollection')}
+                          />
+                          {!!formik.values.optimizedKustomizeManifestCollection && (
+                            <Layout.Horizontal
+                              flex={{ justifyContent: 'flex-start', alignItems: 'center' }}
+                              margin={{ left: 'xlarge' }}
+                              width={430}
+                            >
+                              <FormInput.MultiTextInput
+                                style={{ flex: 1 }}
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/KustomizeWithGIT/KustomizeWithGIT.tsx:419]: inline style?

File: src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/KustomizeWithGIT/KustomizeWithGIT.tsx [hunk 3]
```diff
@@ -377,34 +402,72 @@ function KustomizeWithGIT({
...
+                              )}
+                            </Layout.Horizontal>
+                          )}
+                        </Layout.Vertical>
                       )}
-                    </Layout.Horizontal>
+
+                      <Layout.Horizontal flex={{ justifyContent: 'flex-start', alignItems: 'center' }}>
+                        <FormMultiTypeCheckboxField
+                          name="skipResourceVersioning"
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/KustomizeWithGIT/KustomizeWithGIT.tsx:446]: skipResourceVersioning should be inside advanced section, similar to other manifest

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ManifestSource/KustomizeManifestSource/KustomizeManifestSource.tsx [hunk 4]
```diff
@@ -101,6 +107,19 @@ const Content = (props: ManifestSourceRenderProps): React.ReactElement => {
           />
         </div>
       )}
+      {isFieldRuntime(`${manifestPath}.spec.overlayConfiguration.kustomizeYamlFolderPath`, template) && (
+        <div className={css.verticalSpacingInput}>
+          <FormInput.MultiTextInput
+            disabled={isFieldDisabled(`${manifestPath}.spec.overlayConfiguration.kustomizeYamlFolderPath`)}
+            name={`${path}.${manifestPath}.spec.overlayConfiguration.kustomizeYamlFolderPath`}
+            multiTextInputProps={{
+              expressions,
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ManifestSource/KustomizeManifestSource/KustomizeManifestSource.tsx:116]: Add coverage for this

**Review by bhavya-sinha — CHANGES_REQUESTED** [review]

**Review by bhavya-sinha — APPROVED** [review]

File: src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/KustomizeWithGIT/KustomizeWithGIT.tsx [hunk 5]
```diff
@@ -97,7 +102,9 @@ function KustomizeWithGIT({
           typeof initialValues?.spec?.patchesPaths === 'string'
             ? initialValues?.spec?.patchesPaths
             : initialValues?.spec?.patchesPaths?.map((path: string) => ({ path, uuid: uuid(path, nameSpace()) })),
-        skipResourceVersioning: initialValues?.spec?.skipResourceVersioning
+        skipResourceVersioning: initialValues?.spec?.skipResourceVersioning,
+        optimizedKustomizeManifestCollection: !!kustomizeYamlFolderPath,
```
[the remark this answers is not in the document]
Comment (rameshx) [reply src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/KustomizeWithGIT/KustomizeWithGIT.tsx:106]: `optimizedKustomizeManifestCollection` is not added to the YAML, its value is based on `kustomizeYamlFolderPath` as the user can only give a path if `optimizedKustomizeManifestCollection` is true and vice-versa.

**Review by bhavya-sinha — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Comment by rameshx** [comment]
trigger cypress

**Pull request merged**